DetectorDescription/RecoGeometry/interface/DTGeometryBuilder.h

```diff
@@ -0,0 +1,53 @@
+#ifndef GEOMETRY_RECO_GEOMETRY_DT_GEOMETRY_BUILDER_H
+#define GEOMETRY_RECO_GEOMETRY_DT_GEOMETRY_BUILDER_H
+
+namespace cms {
+
+  class DTGeometryBuilder {
+  public:
+    DTGeometryBuilder() {}
+    ~DTGeometryBuilder() {}
+    
+    using Detector = dd4hep::Detector;
+    using DD3Vector = ROOT::Math::DisplacementVector3D<ROOT::Math::Cartesian3D<double>>;
+
+    void build(DTGeometry&,
+               const DDDetector*, 
+               const MuonNumbering&,
+	       const DDSpecParRefMap&);
+  private:
+    void buildGeometry(DDFilteredView&, dd4hep::Volume top,
+		       const DDSpecPar&,
+		       DTGeometry&, const MuonNumbering&) const;
+    
+    /// create the chamber
+    DTChamber* buildChamber(const DDFilteredView&,
```

Comment:
If they are being created, could this be change to return `std::unique_ptr<DTChamber>`?